libweb3jsonrpc/Debug.cpp

```diff
@@ -136,7 +142,15 @@ Json::Value Debug::debug_storageRangeAt(string const& _blockHashOrNumber, int _t
 		Block block = m_eth.block(blockHash(_blockHashOrNumber));
 
 		unsigned const i = ((unsigned)_txIndex < block.pending().size()) ? (unsigned)_txIndex : block.pending().size();
-		State state = block.fromPending(i);
+		State state(block.state());
+		u256 const rootHash = block.stateRootBeforeTx(_txIndex);
+		if (rootHash)
```

Comment:
Extracted this to common function `dev::eth::createIntermediateState` in State.cpp